src/flags.ts

```diff
@@ -102,64 +72,75 @@ export function isPrecedingCharStringDelimeter(document: vscode.TextDocument, po
 
 const candidateTextStartLocation = (char: number) => (char === 1 ? 0 : char - 2);
 
-interface IFlagManager {
-	store: LDFeatureStore;
-	updateProcessor: LDStreamProcessor;
+async function openFlagInBrowser(key: string) {
+	const flag = await api.getFeatureFlag(config.project, key, config.env);
+
+	// Default to first environment
+	let env: Environment = Object.values(flag.environments)[0];
+	let sitePath = env._site.href;
+
+	if (!config.env) {
+		window.showWarningMessage('[LaunchDarkly] env is not set. Falling back to first environment.');
+	} else if (!flag.environments[config.env]) {
+		window.showWarningMessage(
+			`[LaunchDarkly] Configured environment '${config.env}' has been deleted. Falling back to first environment.`,
+		);
+	} else {
+		env = flag.environments[config.env];
+		sitePath = env._site.href;
+	}
+	opn(url.resolve(config.baseUri, sitePath));
 }
 
-export class LDFlagManager implements IFlagManager {
-	store = InMemoryFeatureStore();
+class FlagStore {
+	store: LDFeatureStore;
 	updateProcessor: LDStreamProcessor;
-	private settings: IConfiguration;
-
-	constructor(ctx: vscode.ExtensionContext, settings: IConfiguration) {
-		this.settings = Object.assign({}, settings);
-		let config = this.config(settings);
-		if (!settings.sdkKey) {
-			console.warn('LaunchDarkly sdkKey is not set. Language support is unavailable.');
-			return;
-		}
 
-		this.updateProcessor = StreamProcessor(settings.sdkKey, config, Requestor(settings.sdkKey, config));
+	constructor() {
+		this.store = InMemoryFeatureStore();
 		this.start();
 	}
 
+	reload(e: ConfigurationChangeEvent) {
+		if (['sdkKey', 'baseUri', 'streamUri'].some(option => e.affectsConfiguration(`launchdarkly.${option}`))) {
```

Comment:
Cleaner logic for checking if we need to reload the store, which also gives us the benefit of not having to pass around configuration settings

Old method is down here: https://github.com/launchdarkly/ld-vscode/pull/26/files#diff-53649f4ba3d51e10eff913927cc17e0cL144